podpac/datalib/smap.py

```diff
@@ -114,16 +119,53 @@ def np2smap_date(date):
         date = str(date).replace('-', '.')
     return date
 
+def _get_from_url(url, auth_session):
+    """Helper function to get data from an NSIDC url with error checking. 
+    
+    Parameters
+    -----------
+    url: str
+        URL to website
+    auth_session: podpac.core.authentication.EarthDataSession
+        Authenticated EDS session
+    """
+    r = auth_session.get(url)
+    if r.status_code != 200:
+        r = auth_session.get(url.replace('opendap/', ''))
+        if r.status_code != 200:
+            raise RuntimeError('HTTP error: <%d>\n' % (r.status_code)
+                               + r.text[:256])
+    return r
+
+
+def _infer_SMAP_product_version(product, base_url, auth_session):
+    """Helper function to automatically infer the version number of SMAP 
+    products in case user did not specify a version, or the version changed
+    
+    Parameters
+    ------------
+    product: str
+        Name of the SMAP product (e.g. one of SMAP_PRODUCT_DICT.keys())
+    base_url: str
+        URL to base SMAP product page
+    auth_session: podpac.core.authentication.EarthDataSession
+        Authenticated EDS session
+    """
+    r = _get_from_url(base_url, auth_session)
+    m = re.search(product, r.text)
+    return int(r.text[m.end() + 1: m.end() + 4])
+
+
 # NOTE: {rdk} will be substituted for the entry's 'rootdatakey'
 SMAP_PRODUCT_DICT = {
```

Comment:
Removed version numbers from SMAP products. The version number is now inferred from the website, although the user could still specify it.